x-pack/plugins/reporting/server/types.ts

```diff
@@ -54,13 +55,15 @@ export interface TimeRangeParams {
   max: Date | string | number | null;
 }
 
+// the "raw" data coming from the client, unencrypted
 export interface JobParamPostPayload {
   timerange: TimeRangeParams;
 }
 
+// the pre-processed, encrypted data ready for storage
 export interface ScheduledTaskParams<JobParamsType> {
```

Comment:
I kind of wish these interfaces are named `JobParamsRaw`, `JobParamsEncrypted`. That's the real difference and makes it less hidden that the shape of a "raw" object can changed when it gets converted to "encrypted"